tracking/tracking_ocp.py

```diff
@@ -0,0 +1,178 @@
+            List of the array of joint velocities.
+            Those velocities were computed using Kalman filter
+            They are used as initial guess
+        markers_ref: List
+            The list of Markers position at each shooting points
+        ocp: OptimalControlProgram
+
+       Methods
+       -------
+       prepare_ocp(self) -> OptimalControlProgram:
+           Prepare the ocp to solve
+       plot_c3d_markers_and_model_markers(self, c3d_path):
+           Plot the positions of each Markers before and after interpolation
+       """
+
+    def __init__(self,
+                 with_floating_base: bool,
+                 c3d_path: str,
+                 n_shooting_points: int,
+                 nb_iteration: int,
+                 ode_solver: OdeSolver = OdeSolver.RK4(),
+                 final_time: float = None,
+                 ):
+        self.with_floating_base = with_floating_base
+        model_path_without_floating_base = "../models/" \
+                                           "wu_converted_definitif_without_floating_base_and_thorax_markers.bioMod"
+        model_path_with_floating_base = "../models/wu_converted_definitif.bioMod"
+        self.ode_solver = ode_solver
+        self.model_path = model_path_with_floating_base if self.with_floating_base \
+            else model_path_without_floating_base
+        self.biorbd_model = biorbd.Model(self.model_path)
+
+        self.q_file = os.path.splitext(c3d_path)[0] + "_q.txt"
+        self.qdot_file = os.path.splitext(c3d_path)[0] + "_qdot.txt"
+
+        self.data = C3dData(c3d_path, self.biorbd_model)
+
+        self.final_time = self.data.get_final_time() if final_time is None else final_time
+        self.n_shooting_points = n_shooting_points
+        self.nb_iteration = nb_iteration
+        self.data_loaded = LoadData(self.biorbd_model, c3d_path, self.q_file, self.qdot_file)
+
+        self.q_ref, self.qdot_ref, self.markers_ref = self.data_loaded.get_experimental_data(
+                                                                                [self.n_shooting_points],
+                                                                                [self.final_time],
+                                                                                with_floating_base=with_floating_base)
+        self.ocp = None
+        self.prepare_ocp()
+
+    def prepare_ocp(self):
+        nb_q = self.biorbd_model.nbQ()
+        nb_qdot = self.biorbd_model.nbQdot()
+
+        # Add objective functions
+        objective_functions = ObjectiveList()
+        objective_functions.add(ObjectiveFcn.Lagrange.MINIMIZE_CONTROL, key="tau", weight=10)
+        objective_functions.add(ObjectiveFcn.Mayer.TRACK_MARKERS, weight=100000, target=self.markers_ref[0], node=Node.ALL)
+        if self.with_floating_base:
+            objective_functions.add(ObjectiveFcn.Lagrange.MINIMIZE_STATE, weight=10, key="qdot",
+                                    index=[0, 1, 2, 3, 4, 5])
```

Comment:
Remove indexes. I think we can remove them, and the if also. It doesn't matter if we are in a floating base or not.

Reply:
done